### PR TITLE
Add an option named "--continue" for "build"  command to apply templates.

### PR DESCRIPTION
**Pull request by AngryBerryMS**
Currently I test it using a real repo and it works well for landing/hub pages.

Will try to modify the now spec test project to support this in another PR. 

Please help review firstly.

### CODE REVIEW

File: src/docfx/build/Builder.cs [hunk 1]
```diff
@@ -30,6 +30,13 @@ public static bool Run(CommandLineOptions options, Package? package = null)
             var operation = Telemetry.StartOperation("build");
             using var errors = new ErrorWriter(options.Log);
 
+            if (options.Continue)
+            {
+                // Apply templates.
+                ApplyTemplates.Run(errors, options);
+                return errors.HasError;
```
Comment (yufeih) [inline src/docfx/build/Builder.cs:37]: Should also call line 46/48, consider not return here.

File: src/docfx/template/TemplateEngine.cs [hunk 2]
```diff
@@ -60,6 +87,20 @@ internal class TemplateEngine
             _searchIndexBuilder = searchIndexBuilder;
         }
 
+        private TemplateEngine(ErrorBuilder errors, Config config, Package package, string locale)
```
Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs:90]: Is it possible to have only one private constructor to remove the duplicated logic?

File: src/docfx/template/TemplateEngine.cs [hunk 3]
```diff
@@ -28,19 +28,46 @@ internal class TemplateEngine
...
+            ErrorBuilder errors,
+            Config? config,
+            PackageResolver? packageResolver,
+            string locale,
+            Package? package,
+            BookmarkValidator? bookmarkValidator = null,
+            SearchIndexBuilder? searchIndexBuilder = null,
+            bool fullBuild = true)
+        {
+            if (fullBuild)
```
Comment (yufeih) [inline src/docfx/template/TemplateEngine.cs:41]: Consider use two static factory methods over a bool switch. With 2 static factory methods, we could remove these null checks.

File: src/docfx/build/apply-templates/ApplyTemplates.cs [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+                DryRun = false,
+                Template = new PackagePath(options.Template),
+            };
+
+            var fileResolver = new FileResolver(package);
+            var jsonSchemaLoader = new JsonSchemaLoader(fileResolver);
+            var jsonSchemaProvider = new JsonSchemaProvider(config, package, jsonSchemaLoader);
+            var templateEngine = TemplateEngine.CreateTemplateEngine(errors, config, null, locale, package, fullBuild: false);
+
+            Parallel.ForEach(
```
Comment (yufeih) [inline src/docfx/build/apply-templates/ApplyTemplates.cs:39]: Use `ParallelUtility.ForEach`, it has logic to handle exceptions.

File: src/docfx/build/apply-templates/ApplyTemplates.cs [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+                File.WriteAllText(GetOutPathWithDifferentExtension(inputDir, filePath, outputDir, "json"), JsonUtility.Serialize(model));
+            }
+            else
+            {
+                var isContentRenderType = jsonSchemaProvider.GetRenderType(ContentType.Page, new SourceInfo<string?>(schema)) == RenderType.Content;
+                if (isContentRenderType)
+                {
+                    var (model, _) = templateEngine.CreateTemplateModel(file, schema is null ? string.Empty : schema.ToString(), pageModel);
+
+                    File.WriteAllText(GetOutPathWithDifferentExtension(inputDir, filePath, outputDir), JsonUtility.Serialize(model));
```
Comment (yufeih) [inline src/docfx/build/apply-templates/ApplyTemplates.cs:73]: Create output directory before calling `File.WriteAllText`

File: src/docfx/build/Builder.cs (hunk 1 above)
Comment (AngryBerryMS) [reply src/docfx/build/Builder.cs]: Fixed.

File: src/docfx/template/TemplateEngine.cs (hunk 3 above)
Comment (AngryBerryMS) [reply src/docfx/template/TemplateEngine.cs]: Done.

File: src/docfx/build/apply-templates/ApplyTemplates.cs (hunk 5 above)
Comment (AngryBerryMS) [reply src/docfx/build/apply-templates/ApplyTemplates.cs:76]: Done

File: src/docfx/build/apply-templates/ApplyTemplates.cs (hunk 4 above)
Comment (AngryBerryMS) [reply src/docfx/build/apply-templates/ApplyTemplates.cs]: Done

File: src/docfx/template/TemplateEngine.cs (hunk 2 above)
Comment (AngryBerryMS) [reply src/docfx/template/TemplateEngine.cs]: Done

**Review by yufeih — APPROVED** [review]

**Review by mingwli — APPROVED** [review]
:shipit:

**Pull request merged**